### ISSUE TITLE
Implement new `Corrector` API design

### ISSUE THREAD

**Issue by barentsen** [KeplerGO/lightkurve#319]
During a brainstorm today (by @christinahedges, @nksaunders, @barentsen), we provisionally decided to deprecate `LightCurve.correct()` in favor of an object-oriented approach centered on an abstract `Corrector` class which has three key methods:
```
Corrector(**data_required):
    .correct(**options) -> LightCurve
    .diagnose(**options) -> Figure
    .interact() -> Bokeh widget
```

Specific implementations of this class would include `SFFCorrector`, `PLDCorrector`, and `PCACorrector`.  The rationale behind removing `lc.correct()` is three-fold:
1. Different correction methods require different input data and different options, which is difficult to expose as a single method.
2. Forcing pixel-level decomposition (PLD) methods to operate on a LightCurve objects is awkward because the method requires pixel-level data. 
3. By exposing `Corrector` class objects, we will encourage users to research the correction methods being applied and, if necessary, develop their own `Corrector` subclass.

In summary, the old syntax:
```Python
lc_corrected = lc.correct(method="sff", **options)
```
...will be replaced by the new syntax:
```Python
lc_corrected = SFFCorrector(lc, centroids).correct(**options)
```

**Comment by barentsen** [comment]
I agree that @gully's comment regarding the loss of the ability to chain is a shame, but calling everything `correct()` is going to be confusing.  How about the following convenience functions: `tpf.pld()`, `lc.sff()`, `lc.cbv()`.

The tutorials should advertise the `Corrector()` class API because we'll want to encourage users to carefully assess and diagnose detrending, which is what that API intends to enable.

**Comment by barentsen** [comment]
Decision reached today in the GO office!  We will enable chaining as follows:
```python
lc = tpf.to_corrector(method="pld").correct()
```
or
```python
lc = lc.to_corrector(method="sff").correct()
```

**Comment by nksaunders** [comment]
@barentsen with #305 and #417 merged in, can this be closed?

**Comment by barentsen** [comment]
I'd like to leave this open until we have added a `diagnose()` method to both SFFCorrector and PLDCorrector.  (Even if all it does for now is return an informative summary plot.)

**Comment by barentsen** [comment]
Closing this issue due to inactivity.  Improvements on the `Corrector` classes continue to be made, e.g. over in #525.
